spaghetti-core/src/main/antlr/Module.g4

```diff
@@ -44,7 +44,9 @@ typeParameters : '<' Name ( ',' Name )* '>'
 	;
 
 structDefinition : ( documentation = Doc )? annotations?
-	'struct' Name typeParameters? '{'
+	'struct' Name typeParameters?
+	( 'extends' superTypeDefinition )?
+	'{'
```

Comment:
On the grammar level, you don't have information about what the `super` type is -- it might not even be defined in the same file. So checking the type of the super-type is handled on a higher level:

https://github.com/prezi/spaghetti/blob/generator-test-support/spaghetti-core/src/main/java/com/prezi/spaghetti/ast/internal/parser/StructParser.java#L68-68